src/main/msp/msp_protocol_v2_sensor_msg.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of INAV Project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * Alternatively, the contents of this file may be used under the terms
+ * of the GNU General Public License Version 3, as described below:
+ *
+ * This file is free software: you may copy, redistribute and/or modify
+ * it under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+#pragma once
+
+typedef struct __attribute__((packed)) {
+    uint8_t quality;    // [0;255]
+    int32_t motionX;
+    int32_t motionY;
+} mspSensorOpflowDataMessage_t;
+
+typedef struct __attribute__((packed)) {
+    uint8_t quality;    // [0;255]
+    int32_t distanceMm; // Negative value for out of range
+} mspSensorRangefinderDataMessage_t;
+
+typedef struct __attribute__((packed)) {
+    uint8_t  instance;                  // sensor instance number to support multi-sensor setups
+    uint16_t gpsWeek;                   // GPS week, 0xFFFF if not available
+    uint32_t msTOW;
+    uint8_t  fixType;
+    uint8_t  satellitesInView;
+    uint16_t horizontalPosAccuracy;     // [cm]
+    uint16_t verticalPosAccuracy;       // [cm]
+    uint16_t horizontalVelAccuracy;     // [cm/s]
+    uint16_t hdop;
+    int32_t  longitude;
+    int32_t  latitude;
+    int32_t  mslAltitude;       // cm
+    int32_t  nedVelNorth;       // cm/s
+    int32_t  nedVelEast;
+    int32_t  nedVelDown;
+    uint16_t groundCourse;      // deg * 100, 0..36000
```

Comment:
we could drop groundCourse as it can be calculated from the velocity, completely up to you if you want to keep it or not. I've changed ArduPilot driver not to use it

Reply:
Let's keep it for simplicity.